src/frontends/pytorch/src/op/atan2.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/atan.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/divide.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/greater.hpp"
+#include "openvino/op/greater_eq.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/logical_and.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/subtract.hpp"
+#include "pt_framework_node.hpp"
+#include "utils.hpp"
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_atan2(const NodeContext& context) {
+    // Check whether inputs present
+    num_inputs_check(context, 2, 3);
+    // "aten::atan2.out(Tensor input,Tensor other, *,Tensor(a!) out) → Tensor(a!)"
+    Output<Node> y;
+    Output<Node> x;
+    // tie inputs together
+    std::tie(y, x) = get_inputs_with_promoted_types(context, 0, 1);
+    auto dummy_const = context.mark_node(ov::op::v0::Constant::create(element::f32, Shape({}), {0.5}))->output(0);
+    // align input types
+    align_eltwise_input_types(context, x, dummy_const, false, true);
```

Comment:
What you want to achieve with this? You called `get_inputs_with_promoted_types` which will promote types to closest common, which can be integer if both were integer. If you want to promote all to closest float type, then you would need to switch the order of calls: first call `align_eltwise_input_types` to x and dummy fp32 const and then call `get_inputs_with_promoted_types` to promote x and y to closest common type.

Reply:
I don't think this solution would work - `get_inputs_with_promoted_types` collects input using `get_input(...)` but doesn't mutate/overwrite them as far as I'm aware, so `align_eltwise_input_types` outcome would probably be ignored. 
Instead, adding `ConvertLike(y, x)` to this code should fix it. Another approach would be one you mentioned, but with second `align_eltwise_input_types(context, y, x, is_python_scalar_input(context, 0), is_python_scalar_input(context, 1)` instead `get_inputs_with_promoted_types`